ctapipe/image/muon/features.py

```diff
@@ -55,15 +52,14 @@ def photon_ratio_inside_ring(
         width of the ring
     """
 
-    total = np.sum(weights)
-
     pixel_r = np.sqrt((center_x - pixel_x)**2 + (center_y - pixel_y)**2)
     mask = np.logical_and(
         pixel_r >= radius - 0.5 * width,
         pixel_r <= radius + 0.5 * width
     )
 
-    inside = np.sum(weights[mask])
+    inside = weights[mask].sum()
+    total = weights.sum()
```

Comment:
Is this with no pixel selection at all? It will be a very noisy quantity. If the goal is to test for possible shower contamination, it is better to take a wider area around the ring (say, 20% of the ring radius or so) and check for pixels above some threshold (or simply post-cleaning) which are outside of that area.

Reply:
This function is agnostic of that, it only does what you see here. You can pass x,y, weights of anything, either a full image or only the pixels after cleaning.

Reply:
Ah, ok, you are right, here you pass x, y so no problem, they can be masked as well. But the comment holds for the higher level call of MuonIntensityFitter, right?

Reply:
Yes, this one should support a mask.